atomic_reactor/core.py

```diff
@@ -280,29 +283,131 @@ def export(self, container, chunk_size=1024*2048):
         return retry(self._export, container, chunk_size, retry=self.retry_times)
 
 
-class DockerTasker(LastLogger):
+class ContainerTasker(LastLogger):
     def __init__(self, base_url=None, retry_times=DOCKER_MAX_RETRIES,
                  timeout=120, **kwargs):
+        self.build_method = None
+        self.base_url = base_url
+        self.retry_times = retry_times
+        self.timeout = timeout
+        self._tasker = None
+        self.kwargs = kwargs
+
+        super(ContainerTasker, self).__init__(**kwargs)
+
+    @property
+    def tasker(self):
+        if self._tasker is None:
+            if self.build_method is None:
+                raise AttributeError("Container task type not yet decided")
+
+            if self.build_method in (CONTAINER_IMAGEBUILDER_BUILD_METHOD,
+                                     CONTAINER_DOCKERPY_BUILD_METHOD):
+                self._tasker = DockerTasker(self.base_url, self.retry_times, self.timeout,
+                                            **self.kwargs)
+                logger.info('ContainerTasker will use DockerTasker')
+            else:
+                raise AttributeError('no tasker')
```

Comment:
Maybe this should report exact reason, like "No ContainerTasker available for build_method %s"

Reply:
changed